<?xml version="1.0" ?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="firefighter">
    <xacro:property name="width" value=".2" />
    <link name="world" />
    <joint name="base_joint" type="fixed">
        <parent link="world" />
        <child link="joint1" />
    </joint>
    <link name="joint1">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04  1.5708 3.14159-->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint1.dae" />
            </geometry>
            <origin xyz="0.0 0 0 " rpy=" 0 0 -1.5708" />
        </visual>
        <collision>
            <geometry>
                <!--- 0.0 0 -0.04  1.5708 3.14159-->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint1.dae" />
            </geometry>
            <origin xyz="0.0 0 0 " rpy=" 0 0 -1.5708" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="1000.0" />
            <inertia ixx="100" iyy="100" izz="100" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="joint2">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint2.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.06096 " rpy=" 0 0 -1.5708" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint2.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.06096 " rpy=" 0 0 -1.5708" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="10" />
            <inertia ixx="1" iyy="1" izz="1" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="joint3">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint3.dae" />
            </geometry>
            <origin xyz="0.0 0 0.03256 " rpy=" 0 -1.5708 0" />
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint3.dae" />
            </geometry>
            <origin xyz="0.0 0 0.03256 " rpy=" 0 -1.5708 0" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="5" />
            <inertia ixx="0.5" iyy="0.5" izz="0.5" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="joint4">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint4.dae" />
            </geometry>
            <origin xyz="0.0 0 0.03056 " rpy=" 0 -1.5708 0" />
        </visual>
        <collision>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint4.dae" />
            </geometry>
            <origin xyz="0.0 0 0.03056 " rpy=" 0 -1.5708 0" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="5" />
            <inertia ixx="0.5" iyy="0.5" izz="0.5" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="joint5">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint5.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.03356 " rpy=" -1.5708 0 0" />
        </visual>
        <collision>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint5.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.03356 " rpy=" -1.5708 0 0" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="2" />
            <inertia ixx="0.2" iyy="0.2" izz="0.2" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="joint6">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint6.dae" />
            </geometry>
            <origin xyz="0 0.00 -0.038 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint6.dae" />
            </geometry>
            <origin xyz="0 0.00 -0.038 " rpy=" 0 0 0" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="2" />
            <inertia ixx="0.2" iyy="0.2" izz="0.2" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="joint6_flange">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint7.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/joint7.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.012 " rpy=" 0 0 0" />
        </collision>
        <selfCollide>true</selfCollide>
        <inertial>
            <mass value="1" />
            <inertia ixx="0.1" iyy="0.1" izz="0.1" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_base">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_base.dae" />
            </geometry>
            <origin xyz="0.0 0 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_base.dae"/> -->
                <cylinder radius="0.02" length="0.01" />
            </geometry>
            <origin xyz="0.0 0 0.01 " rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_left1">
        <visual>
            <geometry>
                <!--- 0.0 0 -0.04 -->
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_left1.dae" />
            </geometry>
            <origin xyz="0.039 -0.0133 -0.012 " rpy=" 0 0 0" />
            <!-- <origin xyz = "0.0 0 -0.012 " rpy = " 0 0 0"/> -->
        </visual>
        <collision>
            <geometry>
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_left1.dae"/> -->
                <cylinder radius="0.0034" length="0.005" />
            </geometry>
            <origin xyz="0.0 0.0 -0.008 " rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_left2">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_left2.dae" />
            </geometry>
            <origin xyz="0.005 -0.0195 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_left2.dae"/> -->
                <cylinder radius="0.0034" length="0.005" />
            </geometry>
            <origin xyz="0.00 -0.0 -0.001 " rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_left3">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_left3.dae" />
            </geometry>
            <origin xyz="0.012 0.0025 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_left3.dae"/> -->
                <cylinder radius="0.0034" length="0.005" />
            </geometry>
            <origin xyz="0.0  0 0.0 " rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_right1">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_right1.dae" />
            </geometry>
            <origin xyz="-0.039 -0.0133 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_right1.dae"/> -->
                <cylinder radius="0.0034" length="0.005" />
            </geometry>
            <origin xyz="0.0 0.0 -0.008 " rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_right2">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_right2.dae" />
            </geometry>
            <origin xyz="-0.005 -0.0195 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_right2.dae"/> -->
                <cylinder radius="0.0034" length="0.005" />
            </geometry>
            <origin xyz="0.0 0.0 -0.001 " rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <link name="gripper_right3">
        <visual>
            <geometry>
                <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_right3.dae" />
            </geometry>
            <origin xyz="-0.012 0.0025 -0.012 " rpy=" 0 0 0" />
        </visual>
        <collision>
            <geometry>
                <!-- <mesh filename="package://mycobot_description/urdf/mycobot_280_m5/gripper_right3.dae"/> -->
                <cylinder radius="0.0034" length="0.005" />
            </geometry>
            <origin xyz="0.0 0 0.0" rpy=" 0 0 0" />
        </collision>
        <inertial>
            <mass value="0.0001" />
            <origin xyz="0 0 0" rpy="0 0 0" />
            <inertia ixx="0.03" iyy="0.03" izz="0.03" ixy="0.0" ixz="0.0" iyz="0.0" />
        </inertial>
    </link>
    <joint name="joint2_to_joint1" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-2.93" upper="2.9321" velocity="1" />
        <parent link="joint1" />
        <child link="joint2" />
        <origin xyz="0 0 0.13156" rpy="0 0 0" />
        <dynamics damping="0.1" friction="0.1"/>
    </joint>
    <joint name="joint3_to_joint2" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-2.36" upper="2.3561" velocity="1" />
        <parent link="joint2" />
        <child link="joint3" />
        <origin xyz="0 0 0" rpy="0 1.5708 -1.5708" />
        <dynamics damping="0.1" friction="0.1"/>
    </joint>
    <joint name="joint4_to_joint3" type="revolute">
        <axis xyz=" 0 0 1" />
        <limit effort="1000.0" lower="-2.62" upper="2.6179" velocity="1" />
        <parent link="joint3" />
        <child link="joint4" />
        <origin xyz="  -0.1104 0 0   " rpy="0 0 0" />
        <dynamics damping="0.1" friction="0.1"/>
    </joint>
    <joint name="joint5_to_joint4" type="revolute">
        <axis xyz=" 0 0 1" />
        <limit effort="1000.0" lower="-2.53" upper="2.5307" velocity="1" />
        <parent link="joint4" />
        <child link="joint5" />
        <origin xyz="-0.096 0 0.06462" rpy="0 0 -1.5708" />
        <dynamics damping="0.1" friction="0.1"/>
    </joint>
    <joint name="joint6_to_joint5" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-2.88" upper="2.8797" velocity="1" />
        <parent link="joint5" />
        <child link="joint6" />
        <origin xyz="0 -0.07318 0" rpy="1.5708 -1.5708 0" />
        <dynamics damping="0.1" friction="0.1"/>
    </joint>
    <joint name="joint6output_to_joint6" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-3.14" upper="3.14159" velocity="1" />
        <parent link="joint6" />
        <child link="joint6_flange" />
        <origin xyz="0 0.0456 0" rpy="-1.5708 0 0" />
        <dynamics damping="0.1" friction="0.1"/>
    </joint>
    <joint name="joint6output_to_gripper_base" type="fixed">
        <!-- <axis xyz="0 0 1"/>
    <limit effort = "1000.0" lower = "-3.14" upper = "3.14159" velocity = "0"/> -->
        <parent link="joint6_flange" />
        <child link="gripper_base" />
        <origin xyz="0 0 0.034" rpy="1.579 0 0" />
    </joint>
    <joint name="gripper_controller" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.599" upper="0.15" velocity="1" />
        <parent link="gripper_base" />
        <child link="gripper_left3" />
        <origin xyz="-0.012 0.005 0" rpy="0 0 0" />
        <dynamics damping="20.0" friction="0.1"/>
    </joint>
    <joint name="gripper_base_to_gripper_left2" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.8" upper="0.5" velocity="1" />
        <parent link="gripper_base" />
        <child link="gripper_left2" />
        <origin xyz="-0.005 0.027 0" rpy="0 0 0" />
        <mimic joint="gripper_controller" multiplier="1.0" offset="0" />
    </joint>
    <joint name="gripper_left3_to_gripper_left1" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.5" upper="0.5" velocity="1" />
        <parent link="gripper_left3" />
        <child link="gripper_left1" />
        <origin xyz="-0.027 0.016 0" rpy="0 0 0" />
        <mimic joint="gripper_controller" multiplier="-1.0" offset="0" />
    </joint>
    <joint name="gripper_base_to_gripper_right3" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.15" upper="0.7" velocity="1" />
        <parent link="gripper_base" />
        <child link="gripper_right3" />
        <origin xyz="0.012 0.005 0" rpy="0 0 0" />
        <mimic joint="gripper_controller" multiplier="-1.0" offset="0" />
    </joint>
    <joint name="gripper_base_to_gripper_right2" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.5" upper="0.8" velocity="1" />
        <parent link="gripper_base" />
        <child link="gripper_right2" />
        <origin xyz="0.005 0.027 0" rpy="0 0 0" />
        <mimic joint="gripper_controller" multiplier="-1.0" offset="0" />
    </joint>
    <joint name="gripper_right3_to_gripper_right1" type="revolute">
        <axis xyz="0 0 1" />
        <limit effort="1000.0" lower="-0.5" upper="0.5" velocity="1" />
        <parent link="gripper_right3" />
        <child link="gripper_right1" />
        <origin xyz="0.027 0.016 0" rpy="0 0 0" />
        <mimic joint="gripper_controller" multiplier="1.0" offset="0" />
    </joint>
   <transmission name="trans_joint2_to_joint1">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint2_to_joint1">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint2_to_joint1_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint3_to_joint2">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint3_to_joint2">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint3_to_joint2_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint4_to_joint3">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint4_to_joint3">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint4_to_joint3_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint5_to_joint4">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint5_to_joint4">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint5_to_joint4_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint6_to_joint5">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint6_to_joint5">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint6_to_joint5_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_joint6output_to_joint6">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint6output_to_joint6">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint6output_to_joint6_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <transmission name="trans_gripper_controller">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="gripper_controller">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        </joint>
        <actuator name="gripper_controller_motor">
            <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>
    <gazebo>
         <plugin filename="libroboticsgroup_upatras_gazebo_mimic_joint_plugin.so" name="mimic_gripper_base_to_gripper_left2">
            <joint>gripper_controller</joint>
            <mimicJoint>gripper_base_to_gripper_left2</mimicJoint>
            <multiplier>1.0</multiplier>
            <offset>0.0</offset>
        </plugin>
        <plugin filename="libroboticsgroup_upatras_gazebo_mimic_joint_plugin.so" name="mimic_gripper_left3_to_gripper_left1">
            <joint>gripper_controller</joint>
            <mimicJoint>gripper_left3_to_gripper_left1</mimicJoint>
            <multiplier>-1.0</multiplier>
            <offset>0.0</offset>
        </plugin>
        <plugin filename="libroboticsgroup_upatras_gazebo_mimic_joint_plugin.so" name="mimic_gripper_base_to_gripper_right3">
            <joint>gripper_controller</joint>
            <mimicJoint>gripper_base_to_gripper_right3</mimicJoint>
            <multiplier>-1.0</multiplier>
            <offset>0.0</offset>
        </plugin>
        <plugin filename="libroboticsgroup_upatras_gazebo_mimic_joint_plugin.so" name="mimic_gripper_base_to_gripper_right2">
            <joint>gripper_controller</joint>
            <mimicJoint>gripper_base_to_gripper_right2</mimicJoint>
            <multiplier>-1.0</multiplier>
            <offset>0.0</offset>
        </plugin>
        <plugin filename="libroboticsgroup_upatras_gazebo_mimic_joint_plugin.so" name="mimic_gripper_right3_to_gripper_right1">
            <joint>gripper_controller</joint>
            <mimicJoint>gripper_right3_to_gripper_right1</mimicJoint>
            <multiplier>1.0</multiplier>
            <offset>0.0</offset>
        </plugin>
      <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>
    <gazebo reference="g_base">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint1">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint2">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint3">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint4">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint5">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint6">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="joint6_flange">
        <selfCollide>true</selfCollide>
    </gazebo>
    <gazebo reference="gripper_left1">
      <disable_collisions_with>gripper_right1</disable_collisions_with>
      <surface>
        <bounce>
          <restitution_coefficient>0.0</restitution_coefficient>
          <threshold>1.0</threshold>
        </bounce>
        <friction>
          <ode>
            <mu>20.0</mu>
            <mu2>20.0</mu2>
          </ode>
        </friction>
      </surface>
    </gazebo>

    <gazebo reference="gripper_left2">
      <disable_collisions_with>gripper_right2</disable_collisions_with>
      <surface>
        <bounce>
          <restitution_coefficient>0.0</restitution_coefficient>
          <threshold>1.0</threshold>
        </bounce>
        <friction>
          <ode>
            <mu>20.0</mu>
            <mu2>20.0</mu2>
          </ode>
        </friction>
      </surface>
    </gazebo>

    <gazebo reference="gripper_left3">
      <disable_collisions_with>gripper_right3</disable_collisions_with>
      <surface>
        <bounce>
          <restitution_coefficient>0.0</restitution_coefficient>
          <threshold>1.0</threshold>
        </bounce>
        <friction>
          <ode>
            <mu>20.0</mu>
            <mu2>20.0</mu2>
          </ode>
        </friction>
      </surface>
    </gazebo>
</robot>

